src/twisted/conch/ssh/filetransfer.py

```diff
@@ -816,35 +816,45 @@ def setAttrs(self, attrs):
         data = self.handle + self.parent._packAttributes(attrs)
         return self.parent._sendRequest(FXP_FSTAT, data)
 
+
+
 class ClientDirectory:
 
     def __init__(self, parent, handle):
         self.parent = parent
         self.handle = NS(handle)
         self.filesCache = []
 
+
     def read(self):
         d = self.parent._sendRequest(FXP_READDIR, self.handle)
         return d
 
+
     def close(self):
         return self.parent._sendRequest(FXP_CLOSE, self.handle)
 
+
     def __iter__(self):
         return self
 
-    def next(self):
+
+    def __next__(self):
         if self.filesCache:
             return self.filesCache.pop(0)
         d = self.read()
         d.addCallback(self._cbReadDir)
         d.addErrback(self._ebReadDir)
         return d
 
+    next = __next__
```

Comment:
`_ebReadDir` modifies the `next` attribute at the end of the listing; it will have to either modify `__next__` as well or one of these should call the other